apps/api/src/app/integrations/usecases/update-integration/update-integration.usecase.ts

```diff
@@ -201,7 +203,7 @@ export class UpdateIntegration {
         active: !!command.active,
       });
 
-      updatePayload.primary = primary;
+      updatePayload.primary = command.conditions && command.conditions.length > 0 ? false : primary;
```

Comment:
What about a case that the primary received true (from the calculate func) => as I understand it primary will also always have the highest priority. 
So could that state happen here? that we are setting primary to false but still the priority here will be set as the highest

Reply:
I guess so not sure how we should handle it thu... @LetItRock ?

Reply:
maybe in the calculate function itself we can check if its now primary and conditions were added in the current update - and if so, recalculate all priorities ? maybe similar to what we do if the active is changed to inactive

Reply:
@davidsoderberg 
This is not the right place, because this if statement will only execute when the "active" flag changes. 


But what we actually want to do is independent of the "active" flag, same as @ainouzgali is suggesting DoD: 
`if existing integration is marked as primary and conditions are updated (exists) we should remove that primary flag and update the priority for all active integrations`.